File: Count distinct elements in every window.java
class Solution {
    ArrayList<Integer> countDistinct(int arr[], int k) {
        // code here
         int count = 0;
         int n = arr.length;
         ArrayList<Integer> list = new ArrayList<>();
      
         
         for(int i=0; i<=n-k; i++){
        HashSet<Integer> set = new HashSet<>();
             int j = i;
             
             while(j < i+k){
                 set.add(arr[j]);
                 j++;
             }
            list.add(set.size()); 
         }
          
        
        
        return list;
    }
}

File: Count distinct elements in every window.cpp
class Solution {
  public:
    vector<int> countDistinct(vector<int> &arr, int k) {
        int i=0;
        vector<int> res;
        unordered_map<int,int> mp;
        for(int j=0;j<arr.size();j++){
            mp[arr[j]]++;
            if(j-i+1==k){
               res.push_back(mp.size());
               mp[arr[i]]--;
               if(mp[arr[i]]==0)mp.erase(arr[i]);
               i++;
            }
        }
        return res;
    }
};
